Newsletters: add 276 (2023-11-08)

harding opened the pull request:
- [ ] Bitcoin Core PR Review Club @LarryRuane 
- [ ] Other stuffs @harding

bitschmidty commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,202 @@
+to be removed from the mempool.
+Since block validation and relay are performance-critical, it's beneficial
+to reduce the required amount of work -- such as updating the fee estimator.
+{% include functions/details-list.md
```
details-list.md can handle 10 Q&A's as an upper bound. I think its probably best either way to trim the number of questions highlighted this week.

IMHO q3,q5,q6 could be candidates for trimming, but Im not deeply familiar with the PR and review

glozow replied:
I agree q3 and q5 are good candidates for trimming. Would also suggest q11.

bitschmidty reviewed (changes requested):
Pushed a commit to fix malformed params for the PR review.

kanzure commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,224 @@
+
+## News
+
+- **Discontinuation of mailing lists:** administrators for the
```
There will be a forthcoming announcement as to where the bitcoin-dev mailing list might be moving to; could this be reworded as something less than total death?

kanzure replied:
Thank you, looks better now.

ismaelsadeeq commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,230 @@
+  q2="Are there any benefits of the `CBlockPolicyEstimator` being a member
+      of `CTxMempool` and updating it synchronously (the current arrangement)?
+      Are there downsides to removing it?"
+  a2="Synchronous code is simpler and easier to reason about. Also, the fee
+      estimator has more visibility into the entire mempool; a downside is
+      the need to encapsulate all the information needed for fee estimation
+      into a new `TransactionInfo` structure.
+      However, not much information is needed."
```
```suggestion
  a2="Synchronous code is simpler and easier to reason about. Also, the fee
      estimator has more visibility into the entire mempool; a downside is
      the need to encapsulate all the information needed for fee estimation
      into a new `NewMempoolTransactionInfo` structure.
      However, not much information is needed by the fee estimator."
```

glozow commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,230 @@
+  a8="The fee estimator needs to know when transactions are removed from the
+      mempool for any reason, not just when a block is connected.
+      Also, the fee estimator needs a transaction's base fee, and this isn't
+      provided via `MempoolTransactionsRemovedForConnectedBlock` (which provides a `CBlock`).
```
Do you mean
```suggestion
      provided via `BlockConnected` (which provides a `CBlock`).
```
?

glozow commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,230 @@
+rest of the `bitcoind` client.
+It moves fee estimator updates from occurring synchronously during mempool
+updates (transactions being added or removed) to instead occur asynchronously.
+This is a performance improvement.
```
Well, not really. I imagine this is actually slower overall, but we're no longer blocking the really important task of processing blocks with it.

LarryRuane replied:
Ah, I see the distinction, you're right, more CPU cycles will be expended almost for sure, but it's the critical path that's faster. I'll improve this wording.

ismaelsadeeq reviewed (commented):
Thanks @harding @LarryRuane.

vostrnad commented on internal/sources.md:
```diff
@@ -0,0 +1,65 @@
+news makes writing the newsletter each week much more manageable.  It's
+possible for us to read every post to a small or mid-sized discussion
+group.  It's not possible for us to read everything about Bitcoin on
+large social media sites or hundreds or individual blogs.
```
```suggestion
large social media sites or hundreds of individual blogs.
```

bitschmidty commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,230 @@
+      per-transaction information needed for fee estimation."
+  a9="That wouldn't be efficient as there may be several thousand transactions
+      in a block, and copying each one's `CTxMempoolEntry` would be expensive,
+      especially since each entry contains references to the transactions parents
```
```suggestion
      especially since each entry contains references to the transaction's parents
```

LarryRuane commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,230 @@
+  a10="It's the sum of the input values minus the sum of the output values.
+      However, the callback can't access the input values because they're stored
+      in the previous transaction outputs (which the callback doesn't have access to).
+      That's why the base fee is included in the `TransactionInfo` structure."
```
Not sure I understand this suggestion. The `TransactionInfo` struct has a member `m_fee`, but `NewMempoolTransactionInfo` only adds four `bool` members. (I'm looking at the latest commit.) Do you still think it's better to refer here to `NewMempoolTransactionInfo`?

LarryRuane commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,224 @@
+rest of the `bitcoind` client.
+It moves fee estimator updates from occurring synchronously during mempool
+updates (transactions being added or removed) to instead occur asynchronously.
+This is a performance improvement.
```
Great suggestion, I forgot about that. I decided to add it to be part of the answer to the first question, since you mentioned this during the chat, rather than in the notes. (I think of this part of the summary as mainly summarizing the notes rather than the discussion.) But let me know if you feel strongly the other way.

LarryRuane commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,224 @@
+updates (transactions being added or removed) to instead occur asynchronously.
+This is a performance improvement.
+
+When a new block is found, its transactions
+to be removed from the mempool.
```
Good catch, what I wrote isn't even English!

murchandamus commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,210 @@
+[Fee Estimator updates from Validation Interface/CScheduler thread][review club 28368]
+is a PR by Abubakar Sadiq Ismail (ismaelsadeeq)
+that modifies the way the transaction fee estimator data is updated.
+(Fee estimation is used with the node's owner initiates a transaction.)
```
```suggestion
(Fee estimation is used when the node's owner initiates a transaction.)
```

harding commented:
Made suggested edits (thanks @kanzure @glozow @ismaelsadeeq @bitschmidty @vostrnad @murchandamus !), added lede, added merge summaries, and added topic links.  I didn't get a chance to review @LarryRuane's section (thanks Larry!), but it looks like several of y'all did already (thanks!).

Thanks everyone!

vostrnad commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,249 @@
+  information users need if they want to [CPFP][topic cpfp] fee bump a
+  channel mutual close transaction.
+
+- [Eclair #2761][] allows forwarding a limited number [HTLCs][topic
```
```suggestion
- [Eclair #2761][] allows forwarding a limited number of [HTLCs][topic
```

vostrnad replied:
not fixed

LarryRuane commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,249 @@
+  q5="[`BlockConnected`][BlockConnected] and [`NewPoWValidBlock`][NewPoWValidBlock]
+      are different callbacks.
+      Which one is asynchronous and which one is synchronous? How can you tell?"
+  a5="`BlockConnected` is asynchronous; `NewPoWValidBlock`is asynchronous.
```
```suggestion
  a5="`BlockConnected` is asynchronous; `NewPoWValidBlock` is asynchronous.
```

azuchi commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,249 @@
+  q5="[`BlockConnected`][BlockConnected] and [`NewPoWValidBlock`][NewPoWValidBlock]
+      are different callbacks.
+      Which one is asynchronous and which one is synchronous? How can you tell?"
+  a5="`BlockConnected` is asynchronous; `NewPoWValidBlock` is asynchronous.
```
Isn't `NewPoWValidBlock` synchronous rather than asynchronous?

bitschmidty replied:
Took this suggestion

glozow commented on _posts/en/newsletters/2023-11-08-newsletter.md:
```diff
@@ -0,0 +1,249 @@
+release candidates.*
+
+- [Bitcoin Core 26.0rc2][] is a release candidate for the next major
+  version of the predominant full node implementation.  Verified test
+  binaries have not yet been released as of this writing, although we
+  expect them to be published at the preceding URL shortly after
+  publication of the newsletter.  There's a brief overview to [suggested
```
rc2 binaries are up

bitschmidty replied:
Thanks @glozow just updated the text accordingly.

bitschmidty reviewed (commented):
Took a suggestion from @azuchi , reviewed topic entries, small change to Bitcoin Core 26.0rc2 release summary
